src/platform/src/main/java/org/geoserver/platform/resource/Files.java

```diff
@@ -300,7 +305,9 @@ private Files() {
      * @param baseDirectory Optional base directory used to resolve relative file URLs
      * @param url File URL or path relative to data directory
      * @return File indicated by provided URL location
+     * @deprecated use resources.
      */
+    @Deprecated
     public static File url(File baseDirectory, String url) {
```

Comment:
Looks like there are a couple of users of this, and those are failing the QA checks.

Reply:
Hmmm, isn't the whole point of deprecation that you allow transition? Does the QA rule not effectively make it useless?

Reply:
No, it just forces the developers deprecating a method to remove its usage everywhere in the code base. It's the usage that triggers QA, not the annotation on the method. 
This also helps to validate that there is indeed a good migration path, and eventually helps improving the javadoc/instructions about what to do to move away from the deprecated method.